src/main/resources/org/opentripplanner/apis/transmodel/schema.graphql

```diff
@@ -2092,6 +2097,58 @@ input PenaltyForStreetMode {
   timePenalty: DoubleFunction!
 }
 
+"""
+One of the listed stop locations must be visited on-board a transit vehicle or the journey must
+alight or board at the location.
+"""
+input PlanPassThroughViaLocationInput {
+  "The label/name of the location. This is pass-through information and is not used in routing."
+  label: String
+  """
+  A list of stop locations. A stop location can be a quay, a stop place, a multimodal
+  stop place or a group of stop places. It is enough to visit ONE of the locations
+  listed.
+  """
+  stopLocationIds: [String]
+}
+
+"""
+A via-location is used to specifying a location as an intermediate place the router must
+route through. The via-location must be either a pass-through-location or a
+visit-via-location. An on-board "visit" is only allowed for pass-through-via-locations, while
+the visit-via-location can visit a stop-location or a coordinate and specify a
+minimum-wait-time.
+"""
+input PlanViaLocationInput @oneOf {
+  passThrough: PlanPassThroughViaLocationInput
+  visit: PlanVisitViaLocationInput
+}
+
+"""
+A visit-via-location is a physical visit to one of the stops or coordinates listed. An
+on-board visit does not count, the traveler must alight or board at the given stop for
+it to to be accepted. To visit a coordinate, the traveler must walk(bike or drive) to
+the closest point in the street network from a stop and back to another stop to join
+the transit network.
+
+NOTE! Coordinates are NOT supported jet.
```

Comment:
```suggestion
NOTE! Coordinates are NOT supported yet.
```

Reply:
Are you sure that you want to allow clients to define both the coordinates and stopLocationIds in the same object with one being silently used as a fallback?

Reply:
Coordinates are not added for now.